File: php/testcode/benchmark_test_00323.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00323(BenchmarkRequest $req): BenchmarkResponse {
    $filename = basename($req->param('file'));
    $path = '/var/app/files/' . $filename;
    if (is_link($path)) {
        return BenchmarkResponse::badRequest('symlinks not allowed');
    }
    $content = file_get_contents($path);
    return BenchmarkResponse::ok($content);
}

File: rust/testcode/benchmark_test_00323.rs
struct ApiClient {
    base_url: &'static str,
}

impl ApiClient {
    fn fetch_item(&self, id: &str) -> String {
        let url = format!("{}/items/{}", self.base_url, id);
        simulated_reqwest_get(&url)
    }
}

pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let item_id = req.param("id");

    if !item_id.chars().all(|c| c.is_ascii_digit()) {
        return super::shared::BenchmarkResponse::bad_request("ID must be numeric");
    }

    let client = ApiClient { base_url: "https://api.example.com" };

    let _resp = client.fetch_item(&item_id);

    super::shared::BenchmarkResponse::ok(&format!("Fetched item: {}", item_id))
}

fn simulated_reqwest_get(url: &str) -> String {
    format!("Response from {}", url)
}
